src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import 'bulma/css/bulma.css';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
 
-import { TodoList } from './components/TodoList';
+import { Loader } from './components/Loader';
 import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
 import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+import { getFilteredTodos } from './tools/filter';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState(0);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  useEffect(() => {
+    getTodos().then((todosFromAPI => {
+      setTodos(todosFromAPI);
+    }));
+  }, []);
+
+  const handleClickTodo = useCallback((todoId: number) => {
+    setSelectedTodo(todoId);
+  }, []);
+
+  const todo = todos.find(item => item.id === selectedTodo);
+
+  const visibleTodos = useMemo(() => {
+    return getFilteredTodos(todos, query, filter);
+  }, [query, todos, filter]);
+
+  const handleCloseModal = () => {
+    setSelectedTodo(0);
+  };
```

Comment:
Wrap the callback with the `useCallback`, cause you are passing this callback to the another component.